app/views/home/index.html.erb

```diff
@@ -0,0 +1,17 @@
+<div class="row">
+<% @photos.each do |photo| %>
+      <% counter = 0 %>
+      <% if photo.parent_uuid != nil %>
+        <% counter += 1 %>
+        <% if counter % 4 == 0 %>
+          <div = class="row">
+        <% end %>
+        <div class="col-xs-6 col-md-3">
```

Comment:
I like what you did here by making the UI responsible by using 2 columns for extra-small screens and 4 columns for medium-sized screens. However you're wrapping a row for every four columns, which seems to not be necessary. You can fill as much columns you want within a row, this will make bootstrap manage the responsibility for you. For example:

```html
<div class="row">
  <div class="col-md-3">…</div>
  <div class="col-md-3">…</div>
  <div class="col-md-3">…</div>
  <div class="col-md-3">…</div>
  <div class="col-md-3">…</div>
  <div class="col-md-3">…</div>
  <div class="col-md-3">…</div>
  ...
</div>
```

Will show 4 columns in medium-sized screens, pushing the additional columns to the next line. A row here have a different meaning of a row in a table, which will keep every column within the same line.